imap/sync_support.c

```diff
@@ -3284,6 +3185,202 @@ int sync_get_uniqueids(struct dlist *kin, struct sync_state *sstate)
 
 /* ====================================================================== */
 
+#ifdef USE_SIEVE
+static int list_cb(void *rock, struct sieve_data *sdata)
+{
+    strarray_t *list = (strarray_t *) rock;
+    size_t namelen = strlen(sdata->name);
+    char *name = xmalloc(namelen + 2); /* use trailing byte as active flag */
+
+    strcpy(name, sdata->name);
+    name[namelen+1] = sdata->isactive;
+
+    strarray_appendm(list, name);
+
+    return 0;
+}
+
+static int remove_cb(const char *sievedir, const char *fname,
+                     struct stat *sbuf, const char *link_target,
+                     void *rock)
+{
+    strarray_t *list = (strarray_t *) rock;
+    char path[PATH_MAX];
```

Comment:
Valgrind complains later (at line 3228) that `path` "may be used uninitialised in conditional or jump" (paraphrasing), which is how I found that this function exists and discovered the cause of the problem in my next comment below.

I've checked pretty closely, and I'm pretty sure it's _not_ used uninitialised (it's always initialised by an snprintf)... but we don't check the return value from the snprintf to ensure it succeeded, and it might be that _that_ is what valgrind is really complaining about.

I might be more comfortable checking the return values of snprintf in this function, and also checking that `strlen(...)` is greater than BYTECODE_SUFFIX_LEN (so that the subtractions to calculate `namelen` can't underflow).  But I think we only get in here from `sievedir_foreach()`, and perhaps we trust that it will always give us good input.  I haven't tried to audit it.

But,

```suggestion
    char path[PATH_MAX] = {0};
```

is enough to shut it up.

Reply:
I guess it's showing my comments in the order I composed them rather than in the order the commented-on sections appear in the source, so "in my next comment below" is actually "in the one above" 😅